src/app/core/internationalization.module.ts

```diff
@@ -16,16 +100,25 @@ export function translateFactory(http: HttpClient) {
       loader: {
         provide: TranslateLoader,
         useFactory: translateFactory,
-        deps: [HttpClient],
+        deps: [HttpClient, StatePropertiesService, TransferState],
       },
     }),
   ],
 })
 export class InternationalizationModule {
-  constructor(@Inject(LOCALE_ID) lang: string, translateService: TranslateService) {
+  constructor(
+    @Inject(LOCALE_ID) angularDefaultLocale: string,
+    translateService: TranslateService,
+    transferState: TransferState
+  ) {
     registerLocaleData(localeDe);
     registerLocaleData(localeFr);
 
-    translateService.setDefaultLang(lang.replace(/\-/, '_'));
+    let defaultLang = angularDefaultLocale.replace(/\-/, '_');
+    if (transferState.hasKey(SSR_LOCALE)) {
+      defaultLang = transferState.get(SSR_LOCALE, defaultLang);
+    }
+    translateService.setDefaultLang(defaultLang);
+    translateService.use(defaultLang);
```

Comment:
Why is it necessary to directly call `use` with the default language? This has introduced a race condition in SSR.

Reply:
Well, not quite. But it actually has introduced non deterministic behavior in connection with `configuration.effects.ts`. We need to stick with this call here otherwise we get issue with initing the app.